pkg/controller/machine/controller.go

```diff
@@ -82,6 +109,27 @@ func newReconciler(mgr manager.Manager, actuator Actuator) reconcile.Reconciler
 	return r
 }
 
+func stringPointerDeref(stringPointer *string) string {
+	if stringPointer != nil {
+		return *stringPointer
+	}
+	return ""
+}
+
+func predicateUpdateFunc(e event.UpdateEvent) bool {
+	oldObject := e.ObjectOld.(*machinev1.Machine)
+	newObject := e.ObjectNew.(*machinev1.Machine)
+	// Don't queue when phasing so we have control over
+	// reconcile.Result{Requeue: true, RequeueAfter: retryTime, err}
+	oldPhase := stringPointerDeref(oldObject.Status.Phase)
+	newPhase := stringPointerDeref(newObject.Status.Phase)
+	if oldPhase != newPhase {
```

Comment:
I don't think I understand this. Maybe I misunderstood part of the original proposal. So, we don't want to process a reconciliation when, for example, moving from `Provisioned` to `Running`? It seems like what we really want is to not reconcile if *only* the phase has changed.